src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxePsiFieldImpl.java

```diff
@@ -213,4 +206,19 @@ public boolean hasModifierProperty(@HaxePsiModifier.ModifierConstant @NonNls @No
     return this.getModifierList().hasModifierProperty(name);
   }
 
+  @NotNull
+  @Override
+  public SearchScope getUseScope() {
+    final PsiElement parent = getParent();
+    if(parent != null) {
+      if (this instanceof HaxeLocalVarDeclarationPart) {
+        // parent of local var declaration-Part
+        return new LocalSearchScope(parent.getParent());
```

Comment:
I'm growing concerned about code like this.  If we have to expand the tree for some reason, then code that expects a parent of a given type at a certain place, then code like this will break.  And I know I've been taking (and probably writing) code like this a lot.  I think that perhaps we should be using UsefulPsiTreeUtil.getParentOfType() instead of expecting a specific element's tree position.  

What do you think?